internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -216,6 +235,20 @@ func (r *BMACReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (c
 		return result.Result()
 	}
 
+	// To be able to get ignition from the mcs on from spoke cluster, we must have certificate of the mcs and to connect to tls port of mcs
+	result = r.ensureMCSCert(ctx, log, bmh, agent)
+	if result.Dirty() {
+		err := r.Client.Update(ctx, bmh)
+		if err != nil {
+			log.WithError(err).Errorf("Error updating agent")
```

Comment:
We are updating the bmh here. Could you write a mortgage descriptive error message?

Reply:
Done